controller/array_action/array_mediator_ds8k.py

```diff
@@ -522,20 +505,25 @@ def _get_api_snapshot(self, snapshot_name, pool_id=None):
                                                                    self.service_address)
         return api_snapshot
 
-    def get_snapshot(self, snapshot_name, pool_id=None):
-        api_snapshot = self._get_api_snapshot(snapshot_name, pool_id)
+    @convert_scsi_id_to_array_id
+    def get_snapshot(self, volume_id, snapshot_name, pool=None):
+        if not pool:
+            source_api_volume = self._get_api_volume_by_id(volume_id)
+            pool = source_api_volume.pool
+        api_snapshot = self._get_api_snapshot(snapshot_name, pool)
         if api_snapshot is None:
             return None
         return self._generate_snapshot_response_with_verification(api_snapshot)
 
-    def _create_similar_volume(self, target_volume_name, source_api_volume):
+    def _create_similar_volume(self, target_volume_name, source_api_volume, pool_id):
```

Comment:
please choose a parameters order as you see fit and ensure it's aligned with `_create_snapshot`

Reply:
done